### PR TITLE
Management Plane bootstrap

### PR DESCRIPTION
**Pull request by phudtran**
This PR adds the following.
Bootstraps the cluster with default DFTs, Chains, FTNs, and FWDs
Adds the FWD crd object
Bootstraps the kind setup with docker containers to be used as droplets for the management plane.

### CODE REVIEW

File: src/mgmt/store/operator_store.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 import logging
 import inspect
+from common.constants import KIND
```
Comment (liangbin-pub) [inline src/mgmt/store/operator_store.py:25]: Is this tied to kind cluster?

File: src/mgmt/workflows/chains/create.py [hunk 2]
```diff
@@ -6,13 +6,23 @@
...
     logger.info("Creating Chain {}!".format(name))
     if not chain:
         chain = chains_opr.get_stored_obj(name, spec)
+    if not dfts_opr.store.contains_obj(chain.dft, KIND.dft):
+        task.raise_temporary_error(
+            "Parent DFT {} not yet created!".format(chain.dft))
+    ftns_opr.create_n_ftns(chain)
+    chain.tail = chain.ftns[chain.size - 1]
+    chain.head = chain.ftns[0]
     return chain
```
Comment (liangbin-pub) [inline src/mgmt/workflows/chains/create.py:28]: Should have number of chains and replicas decided based on number of ftn droplets

File: src/mgmt/workflows/ftns/create.py [hunk 3]
```diff
@@ -5,12 +5,28 @@
...
+        task.raise_temporary_error(
+            "Parent DFT {} not yet created!".format(ftn.dft))
+    if not chains_opr.store.contains_obj(ftn.parent_chain, KIND.chain):
+        task.raise_temporary_error(
+            "Parent Chain {} not yet created!".format(ftn.parent_chain))
+    if len(droplets_opr.get_unallocated_droplets()) < 1:
+        task.raise_temporary_error(
+            "No droplets available for FTN")
+    droplets_opr.assign_droplet(ftn)
     return ftn
```
Comment (liangbin-pub) [inline src/mgmt/workflows/ftns/create.py:32]: each network node will offer 2 droplets, one is physically connects to tenant network, so can only used as fwd; the other physically connects to ZGC internal network, which can only used for ftn. So droplets do have an attribute (role) already decided by how they physically connected.
For example, when there are 3 nodes allocated for ZGC, we will have eth2(for example) of each node on tenant network, eth3 on ZGC internal network. so total we got 6 droplets, but up to 3 fwds (eth2s) and up to 3 ftns (eth3s)

you can leave it here and we can make adjustment once external API is hooked up.

File: src/mgmt/workflows/chains/create.py (hunk 2 above)
Comment (phudtran) [reply src/mgmt/workflows/chains/create.py:28]: My thought was to have the cluster admin decide on the number of chains and replicas/chain based on the DFT object spec.
From DFT crd definition
```
    - name: Numchains
      type: integer
      priority: 0
      JSONPath: .spec.numchains
      description: The number of chains in the DFT
    - name: Numchainreplicas
      type: integer
      priority: 0
      JSONPath: .spec.numchainreplicas
      description: The number of replica ftns in each chain
```
Zeta can then give a temporary error and retry until the number of required droplets are available.

If we are to have these dynamically allocated based on existing resources, we can remove these fields in the DFT crd definition, and instead set a limit on the minimum number of droplets that's needed for a DFT to be created.
What are your thoughts?

File: src/mgmt/store/operator_store.py (hunk 1 above)
Comment (phudtran) [reply src/mgmt/store/operator_store.py:25]: This is a reference to the Kind field for crd objects
```
apiVersion: "zeta.com/v1"
 kind: Fwd
 metadata:
```

File: src/mgmt/workflows/ftns/create.py (hunk 3 above)
Comment (phudtran) [reply src/mgmt/workflows/ftns/create.py:32]: I see, so there's a restriction on the **type** of droplets that can be used for a FWD or FTN respectively.
Gotcha, lets revisit this when the external API is ready.

**Pull request merged**